nixos/modules/services/networking/nix-serve.nix

```diff
@@ -61,16 +59,19 @@ in
 
       path = [ config.nix.package.out pkgs.bzip2.bin ];
       environment.NIX_REMOTE = "daemon";
-      environment.NIX_SECRET_KEY_FILE = cfg.secretKeyFile;
+
+      script = ''
+        export NIX_SECRET_KEY_FILE="$CREDENTIALS_DIRECTORY/NIX_SECRET_KEY_FILE"
```

Comment:
This also might have to be wrapped in an optional string than if we have no secret file.